### PR TITLE
platform/oci: update tests

### PR DESCRIPTION
**Pull request by arithx**
Blacklists cloudconfig tests, adds an `IsEmpty` function to `platform/conf`, & implements the `authorized_ssh_key` metadata key.

See individual commit messages for context.

### CODE REVIEW

File: platform/api/oci/instance.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ func (a *API) CreateInstance(name, userdata string) (*Machine, error) {
 
 	opts := baremetal.LaunchInstanceOptions{
 		Metadata: map[string]string{
-			"user_data": base64.StdEncoding.EncodeToString([]byte(userdata)),
+			"user_data":           base64.StdEncoding.EncodeToString([]byte(userdata)),
+			"ssh_authorized_keys": sshKey,
```
Comment (bgilbert) [inline platform/api/oci/instance.go]: Do we need to omit this field if `sshKey` is empty?

Comment (arithx) [reply platform/api/oci/instance.go]: Pushed a change to omit it when empty

File: platform/api/oci/instance.go [hunk 2]
```diff
@@ -35,10 +35,15 @@ func (a *API) CreateInstance(name, userdata string) (*Machine, error) {
 		return nil, err
 	}
 
+	metadata := map[string]string{
+		"user_data": base64.StdEncoding.EncodeToString([]byte(userdata)),
```
Comment (bgilbert) [inline platform/api/oci/instance.go]: Come to think of it, could you add another commit which sets this conditionally as well?  Users will probably not explicitly set a zero-length `user_data` field.

Comment (arithx) [reply platform/api/oci/instance.go]: New commit pushed addressing this.

**Comment by arithx** [comment]
Good to merge

**Pull request merged**